### ISSUE TITLE
antsApplyTransforms: incorrect return codes

### ISSUE THREAD

**Issue by janeisklar** [stnava/ANTs#252]
```bash
$ antsApplyTransforms -e 3 -d 3 -i foo.nii -o bar.nii -r baz.nii -t foobar.mat
Transform file does not exist: foobar.mat
$ echo $?
0
```

It would be nice if antsApplyTransforms and potentially other binaries could return proper failure status codes as it is otherwise rather hard to react on errors. Maybe I was just unlucky in this case as I've seen that return EXIT_FAILURE is used throughout the code. I unfortunately don't know the code base well enough to supply a patch for this specific issue, so it would be nice if someone could look into this.

Thank you!

**Comment by cookpa** [comment]
There are many functions that return EXIT_FAILURE, but this is often ignored at the class / executable level, and EXIT_SUCCESS, or literal 0, is returned.

**Comment by gdevenyi** [comment]
I ran into a similar issue with N4BiasFieldCorrection, some of its failures are completely silent without the verbose flag, and the return code is success.

**Comment by ntustison** [comment]
Pull requests are always welcome.

**Comment by ntustison** [comment]
Also, pointing to specific cases (data and usage) is much easier to fix than non-specific claims of something not working.  I've actually gone through the executables to address this precise issue but I could've easily missed things.

**Comment by cookpa** [comment]
Consistent exit codes are helpful for debugging, especially now that the error messages have been curtailed. 

What should be done if the command is invoked with no args? I've seen that classified as success (return  0) and failure (return 1) in various unix contexts. My personal policy is to call this a failure, because it's almost always a sign of failure if that happens in a script, which is where 99% of exit code checks will live.

If "-h" or "--help" is passed, I call that a success. That's unlikely to happen by accident, whereas 

```antsRegistration $args```

with $args being undefined could happen.

So in summary, I propose

EXIT_SUCCESS (not literal 0) on success (EXIT_SUCCESS returned internally to main)
EXIT_FAILURE (not literal 1) on failure (EXIT_FAILURE returned internally to main)

EXIT_SUCCESS if command is run with -h or --help
EXIT_FAILURE if command is run with no args or insufficient args

If that sounds good, I will make a branch to get started on this.

**Comment by ntustison** [comment]
Sounds good to me.  I think your breakdown is what I maintained in my first pass of this issue.

**Comment by cookpa** [comment]
Re opening this now I've made a branch for fixing exit codes:

https://github.com/stnava/ANTs/tree/returnExitCodes

I made one commit, changing antsApplyTransforms. As I thought, the success / failure return values are mostly correctly coded, and it was just a matter of returning the value from the call to `antsApplyTransforms()` rather than returning `EXIT_SUCCESS` after the switch statement.

**Comment by ntustison** [comment]
Hi Phil,  Are you going to make a pull request since everything is passing?

**Comment by cookpa** [comment]
Will do - I was working through the other programs alphabetically but I only got as far as A. I'll leave the branch open so I (or anyone else who feels like it) can come back to it later.

**Comment by ntustison** [comment]
Sounds good.

**Comment by cookpa** [comment]
The above mentioned branch has been merged into master. Most programs, including antsApplyTransforms and N4BiasFieldCorrection, should return non-zero exit codes where appropriate.
